### PR TITLE
Remove Session Manager

### PR DESCRIPTION
**Pull request by DorianGray**

### CODE REVIEW

**Comment by ramnes** [comment]
LGTM to me but do you have any specific reason, other than Dependency Injection is Great™?

**Comment by DorianGray** [comment]
Not really, just cleanup that I had thought about doing while tinkering with the lifecycle changes.

Session manager has no business looking at the config, kore, no_spawn, or state parameters. I'm planning on moving _should_restart into qtile after, have it talk directly to lifecycle instead of a flag. Should I include that in this PR to make it make more sense?

**Comment by DorianGray** [comment]
There, went ahead and finished the related changes.

**Comment by DorianGray** [comment]
got rid of 12 lines =)

File: libqtile/core/manager.py [hunk 1]
```diff
@@ -197,6 +198,16 @@ def _configure(self):
...
         hook.fire("startup_complete")
 
+    @property
+    def display_name(self):
+        # Dots might appear in the host part of the display name
+        # during remote X sessions. Let's strip the host part first
+        name = self.core.display_name
+        display_number = name.partition(":")[2]
+        if "." not in display_number:
+            name += ".0"
```
Comment (m-col) [inline libqtile/core/manager.py:208]: Could you move this part into the core? It is X specific

**Comment by m-col** [comment]
LGTM

File: libqtile/core/manager.py (hunk 1 above)
Comment (DorianGray) [reply libqtile/core/manager.py:225]: I'll need to move it and add an ipc_socket_path property to core as well, is that ok? My reasoning is that when we have a separate core there will be a separate way of naming the socket for the ipc server now managed by Qtile.

Comment (m-col) [reply libqtile/core/manager.py:225]: Actually thinking about it again, this string manipulation is only required when finding the socket file, _not_ when we are using the display of `Qtile` (necessarily). I.e. it is also used by command clients to communicate with `Qtile`. So maybe just put it as a standalone function like it was originally but inside `libqtile/backend/x11/__init__.py` or somewhere like that.

Comment (DorianGray) [reply libqtile/core/manager.py:225]: RE irc: I'll take a look at how the command clients do things and try to emulate them while I try to move this logic into the backend.

**Comment by DorianGray** [comment]
This should fix https://github.com/qtile/qtile/issues/2087 as well

File: libqtile/core/manager.py (hunk 1 above)
Comment (m-col) [reply libqtile/core/manager.py:225]: Actually I think you can completely remove this property and just use `self.core.display_name` on line 211. The lines 223-225 are redundant as they are also within `ipc.find_sockfile`. If we need to change this in the future due to new cores then we can deal with it if/when we need to.

**Comment by DorianGray** [comment]
strangely, CI seems more stable on this branch but I have no idea why.

**Comment by DorianGray** [comment]
not just more stable...passes every time. wat

File: libqtile/core/manager.py [hunk 2]
```diff
@@ -701,8 +724,7 @@ def _focus_by_click(self, e):
 
     def process_button_click(self, button_code, state, x, y, event) -> None:
         self.mouse_position = (x, y)
-        k = self.mouse_map.get(button_code)
-        for m in k:
+        for m in self.mouse_map.get(button_code, []):
```
Comment (DorianGray) [inline libqtile/core/manager.py:727]: For anyone wondering why this change is here... mypy found a bug!

**Review by ramnes — APPROVED** [review]
LGTM

File: libqtile/scripts/start.py [hunk 3]
```diff
@@ -89,7 +90,7 @@ def start(options):
     rename_process()
     q = make_qtile(options)
     try:
-        q.loop()
+        asyncio.run(q.loop())
```
Comment (ramnes) [inline libqtile/scripts/start.py:93]: Can we make the API simpler and keep `loop` as a synchronous function that calls `asyncio.run` itself, as `SessionManager` used to?

**Comment by DorianGray** [comment]
Updated based on @ramnes feedback
